packages/edit-widgets/src/components/layout/unsaved-changes-warning.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useEffect } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Warns the user if there are unsaved changes before leaving the editor.
+ *
+ * This is a duplicate of the component implemented in the editor package.
+ * Duplicated here as edit-widgets doesn't depend on editor.
+ *
+ * @return {WPComponent} The component.
+ */
+export default function UnsavedChangesWarning() {
+	const isDirty = useSelect( ( select ) => {
```

Comment:
Nitpick: `isDirty` looks like a boolean to me, how about renaming it to `getIsDirty` or `checkIsDirty` to indicate it as a function more clearly?